cheatsheets/Clickjacking_Defense_Cheat_Sheet.md

```diff
@@ -311,3 +313,166 @@ Activate [designMode](https://developer.mozilla.org/en-US/docs/Web/API/Document/
+
+## Attack Scenario
+
+1. The attacker loads a malicious webpage containing a transparent iframe overlaying a legitimate website.
+
+2. The user is tricked into clicking an element (e.g., a button or link), thinking it belongs to the attacker’s site.
+
+3. The first click moves the transparent iframe into position over a critical UI element of the target website.
+
+4. The second click executes a malicious action, such as transferring funds, changing security settings, or posting content without user consent.
+
+## Ineffective Defenses
+
+Traditional Clickjacking protections are ineffective against Double Clickjacking, as noted in research by Paulos Yibelo:
+
+### 1. **X-Frame-Options:**
+
+  Attackers manipulate iframe positions dynamically, making this defense ineffective.
+
+### 2. **Content Security Policy (CSP) Frame-Ancestors:**
+
+This protection does not prevent UI manipulation via JavaScript or CSS.
+
+### 3. **SameSite Cookies:**
+
+Attackers do not need cross-site requests, rendering this defense ineffective.
+
+## Advanced Mitigation Strategies
+
+Given the shortcomings of traditional defenses, the following advanced approaches offer stronger protection against Double Clickjacking:
+
+### **1. Detect Hidden Iframes Using Intersection Observer**
+
+**Concept:**  Attackers often use transparent or off-screen iframes. This detects and alerts users if such an iframe is capturing interactions.
+
+```javascript
+const observer = new IntersectionObserver(entries => {
+    entries.forEach(entry => {
+        if (!entry.isIntersecting) {
+            alert("Warning: A hidden iframe may be attempting a Clickjacking attack!");
+        }
+    });
+}, { threshold: 0.5 });
+
+document.querySelectorAll("iframe").forEach(iframe => observer.observe(iframe));
+```
+
+✅ Prevents hidden iframe overlays from capturing clicks.
+
+✅ Ensures users interact only with visible elements.
+
+#### **2. Enforce Click-Only on Trusted Domains**
+
+**Concept:** Attackers host iframes on malicious domains. This script ensures actions only occur if the request comes from a trusted domain.
+
+```javascript
+if (document.referrer && !document.referrer.includes("your-website.com")) {
+    alert("Warning: Clicks from an untrusted source detected!");
+}
+```
+
+✅ Blocks interactions initiated from untrusted sites.
+
+✅ Prevents attacks executed via external embedding.
+
+### **3. Delayed Click Action (Click Delay)**
+
+**Concept:** Prevents rapid unintended double-clicks by enforcing a minimum delay between clicks.
+
+```javascript
+let lastClickTime = 0;
+document.getElementById("submitButton").addEventListener("click", function (event) {
+  const now = Date.now();
+  if (now - lastClickTime < 500) {  // 500ms delay
+    event.preventDefault();
+    alert("Double-click detected! Please click only once.");
+  } else {
+    lastClickTime = now;
+    // Proceed with the action
+  }
+});
+```
```

Comment:
This seems feasible. Have you tried this on Yibelo's PoC on his blog post?
What do you think @jmanico ?